packages/collaboration-extension/src/filebrowser.ts

```diff
@@ -60,6 +67,28 @@ export const defaultFileBrowser: JupyterFrontEndPlugin<IDefaultFileBrowser> = {
       tree,
       labShell
     );
+
+    // Fetch settings if possible.
+    if (settingRegistry) {
+      settingRegistry
+        .load('@jupyterlab/notebook-extension:tracker')
+        .then(settings => {
+          const updateSettings = (settings: ISettingRegistry.ISettings) => {
+            const enableDocWideUndo = settings?.get(
+              'experimentalEnableDocumentWideUndoRedo'
+            ).composite as boolean;
+            const disableDocumentWideUndo = !enableDocWideUndo ?? true;
+            drive.sharedModelFactory.disableDocumentWideUndo =
```

Comment:
What do you think of introducing a generic pattern:

```ts
drive.sharedModelFactory.documentOptions: Map<string, Record<string, any>>
```

The key of the map could be the document type directly instead of string I guess. And at the instantiation of the shared models, you will pass the options.

Something like:

```ts
      case 'notebook':
        sharedModel = new YNotebook(this.documentOptions.get('notebook'));
```

Reply:
Sound good! I'm on it.